File: Arrays/Array-Rearrange/ArrayRearrange.java
public class ArrayRearrange {

    public static void main(String[] args) {
        int arr[] = { -12, 11, -13, -5, 6, -7, 5, -3, -6};
        int left = 0;
        int right = arr.length - 1;
        while(left < right){
            if(arr[left] < 0){
                left++;
            }
            else if(arr[right] > 0){
                right--;
            }
            else{
                arr[left] += arr[right];
                arr[right] = arr[left] - arr[right];
                arr[left] = arr[left] - arr[right];
                left++;
                right--;
            }
        }

        for(int i = 0; i<arr.length; i++){
            System.out.print(arr[i]+" ");
        }
    }
}

File: Arrays/Array-Rearrange/array_rearrange.py
arr = [-12, 11, -13, -5, 6, -7, 5, -3, -6]
left = 0
right = len(arr) - 1

while left < right:
    if arr[left] < 0:
        left += 1
    elif arr[right] > 0:
        right -= 1
    else:
        arr[left], arr[right] = arr[right], arr[left]
        left += 1
        right -= 1

print(arr)